src/plugin/buddhistEra/index.js

```diff
@@ -0,0 +1,21 @@
+import { FORMAT_DEFAULT } from '../../constant'
+
+export default (o, c) => { // locale needed later
+  const proto = c.prototype
+  const oldFormat = proto.format
+  // extend en locale here
+  proto.format = function (formatStr, localeObject) {
+    const yearBias = 543
+    const locale = localeObject || this.$locale()
+    const str = formatStr || FORMAT_DEFAULT
+    const result = str.replace(/BBBB|BB/g, (match) => {
+      switch (match) {
+        case 'BB':
+          return (this.$y + yearBias) % 100
```

Comment:
@saknarak we should check and add the leading zero using `padStart`, consider this case `2000 % 100`